### ISSUE TITLE
Proposal for a "Jobs view" in the history

### ISSUE THREAD

**Issue by neoformit** [galaxyproject/galaxy#15847]
After receiving feedback from a number of Galaxy users (and through my own use) I wonder if there could be a more intuitive layout for the history UI. I have spend some time fleshing out this concept (as requested by @bgruening) but there may be areas of the UI where I haven't considered the implications of this layout - feedback please :)

I think something like this could lead to a much more intuitive, cleaner experience for the history.

**Current implementation:**

- A job produces any (potentially unknown) number of datasets in the history
- Dataset names often become opaque and confusing with long histories
- The state/logic of the job and the dataset are merged into one element
- There is no UI structure to distinguish the output of different jobs
- Histories become long very quickly, and navigation suffers as a result
- The above problems are reflected in the tool form when selecting an input dataset (unclear which is which)
- Per-job operations can be error-prone and tedious (especially with many outputs per-job)
- Renaming of job outputs has to be done retrospectively, one-by-one (again, tedious)

**Proposed implementation:**

- The history lists jobs as the primary element, with datasets nested beneath them (as though a job's outputs are a list collection)
- Job-related state/logic are located in the job element
- Dataset-related state/logic are located in the dataset element
- The history length is identical to the number of jobs that have been run
- The jobs-view allows simple navigation by job - separation of outputs is inherent in the design
- Clicking on a job element displays the output datasets of that job (much like clicking on a collection)
- Per-job operations are clear and quick - one can "delete a job" instead of selecting all datasets one-by-one
- Dataset selection in the tool form is clarified by semantic naming of outputs e.g. `job xyz > dataset 1`
- The job can be named prospectively in the tool form (before the `Run` button)
- Finally, this UI layout might be implemented as a "View" that the user can toggle on/off (or select from multiple). The conventional dataset view may still be preferred in some situations.

### Current implementation

First, an example to demonstrate where the current history layout gets real confusing. Here is a history where I repeated a job on the same dataset with different parameters. It's not clear where one job's outputs end and the next one starts:

![galaxy-history-cluttered](https://user-images.githubusercontent.com/42562517/227822190-fce8e8ac-f6e6-4f1e-a483-6d96ae5b3bf3.png)

Now, a second example I manufactured as a use-case. A user has uploaded 6 raw read files, cleaned them with Trimmomatic and then assembled with Trinity. For comparison, they have also assembled the raw reads. With the current implementation the result is a history spanning multiple pages. To confirm what the final Trinity jobs entail, we have to manually cross-reference `data 1` and `data 7`.

![galaxy-history-mockup-current](https://user-images.githubusercontent.com/42562517/227857193-92310033-251f-439b-a636-ab5abea20bb4.png)

### Proposed implementation

A similar history to the above, but with the proposed "jobs-view" (blue traces follow a click). With a glance at this history, the logic of the analysis is immediately clear:

![galaxy-history-mockup-job-view](https://user-images.githubusercontent.com/42562517/227857239-875393c1-e3db-47ac-8e0b-94e565273559.png)

Viewing job/dataset information:

![galaxy-history-mockup-information](https://user-images.githubusercontent.com/42562517/227854489-772a7b05-dabc-49d8-aee2-9692614d442d.png)

Selecting a dataset and naming a job in the tool form:

![galaxy-history-mockup-tool-form](https://user-images.githubusercontent.com/42562517/227854589-a9f2f0d5-5c4b-4457-b804-cf834a8ebeae.png)

**Comment by nsoranzo** [comment]
I remember discussing ideas along these lines during the hackathon of probably GCC2016, it would be really nice to have! 

> Finally, this UI layout might be implemented as a "View" that the user can toggle on/off (or select from multiple). The conventional dataset view may still be preferred in some situations.

This would be key, especially to not disrupt existing user expectations.

**Comment by mvdbeek** [comment]
I completely agree, this to me seems like a very good and simple enhancement, and was always my first alternative Proposition to a graph view for histories.
